.github/workflows/mattermost.yml

```diff
@@ -0,0 +1,41 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Mattermost_message
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the develop branch
+  push:
+    branches: [ feature-CI_mattermost_messages, develop, main ]
+  pull_request:
+    branches: [ feature-CI_mattermost_messages, develop, main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: push  
+        uses: Mitigram/gh-action-mattermost@main
+        if: github.event_name == 'push'
+        with: 
+          username: t8y
+          url: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
+          text: ${{ github.event.pusher.name }} pushed into ${{ github.event.ref }}. See ${{ github.event.compare }} for more details.
+      - name: pull_request
+        uses: Mitigram/gh-action-mattermost@main
+        if: github.event_name == 'pull_request' && github.event.action != 'synchronize'
+        with:
+          username: t8y
+          url: ${{ secrets.MATTERMOST_WEBHOOK_URL }}
+          text: Pull request ${{ github.event.number }} has been ${{ github.event.action }}.
+      - name: synchronize
```

Comment:
what does synchronize do?
Could you please explain to me and add a comment?

Reply:
ah, so this is what i thought the push above would do.
Maybe we should just see how it is in practice.